[오류] 부산행정 크롤러 로그인 실패 - Alert 처리 문제

## 🐛 부산행정 크롤러 로그인 실패 - Alert 처리 문제

### 문제 설명
부산행정 크롤러에서 로그인 시 발생하는 2개의 연속적인 JavaScript Alert을 제대로 처리하지 못해 크롤링이 계속 실패하고 있습니다.

### 오류 로그
```
2025-06-02T13:04:27.663+09:00 ERROR ... 부산행정 로그인 실패: unexpected alert open: {Alert text : 비밀번호 변경 기간입니다.비밀번호 변경 페이지로 이동합니다.}
...
Caused by: org.openqa.selenium.UnhandledAlertException: unexpected alert open
```

### 현재 상황
- **문제 발생 사이트**: 부산행정 크롤러만 해당 (다른 8개 크롤러는 정상 작동)
- **발생 빈도**: 매 스케줄링마다 발생 (1분 간격)
- **Alert 개수**: 2개 연속 발생
- **Alert 처리 후 이동 페이지**: `https://home.pen.go.kr/helpcenter/lo/login/changePassword.do`

### 현재 동작
1. 로그인 시도
2. 첫 번째 Alert 발생 → 처리됨
3. 두 번째 Alert 발생 → **처리되지 않음**
4. `UnhandledAlertException` 발생으로 로그인 실패

### 예상 해결 방안
1. **`BusanAdminCrawler.handlePopup()` 수정**: 2개의 연속 Alert 처리
2. **로그인 성공 조건 수정**: 비밀번호 변경 페이지도 성공으로 인식

--- a/src/main/java/com/helpcentercrawl/crawler/core/AbstractCrawler.java
+++ b/src/main/java/com/helpcentercrawl/crawler/core/AbstractCrawler.java
@@ -193,6 +193,9 @@ public abstract class AbstractCrawler implements SiteCrawler {
      */
     protected void processData() {
         try {
+            // 데이터 처리 시작 전 Alert 확인
+            handleUnexpectedAlert();
+
             String tableSelector = getTableSelector();
             List<WebElement> rows = driver.findElements(By.cssSelector(tableSelector));
 
@@ -201,8 +204,6 @@ public abstract class AbstractCrawler implements SiteCrawler {
                     List<WebElement> cells = row.findElements(By.tagName("td"));
 
                     if (cells.size() > Math.max(getTitleIndex(), getDateIndex())) {
-
-
                         if (isNotice(cells.get(0))) {
                             continue;
                         }
@@ -219,20 +220,37 @@ public abstract class AbstractCrawler implements SiteCrawler {
                         }
 
                         titleText = titleProcessing(titleText);
-
                         LocalDate date = dateProcessing(dateCell);
-
                         boolean completed = statusProcessing(row);
 
                         saveDataBuild(titleText, date, completed);
                     }
+                } catch (UnhandledAlertException e) {
+                    log.warn("{} - 행 처리 중 Alert 발생, 해당 행 건너뜀: {}", getSiteName(), e.getAlertText());
+                    handleUnexpectedAlert();
                 } catch (Exception e) {
                     log.warn("{} - 행 데이터 처리 중 오류 발생: {}", getSiteName(), e.getMessage());
                 }
             }
 
+        } catch (UnhandledAlertException e) {
+            log.warn("{} - 데이터 처리 중 Alert으로 인한 중단: {}", getSiteName(), e.getAlertText());
+            handleUnexpectedAlert();
         } catch (Exception e) {
             log.error("데이터 처리 오류: {}", e.getMessage());
+        }
+    }
+
+    private void handleUnexpectedAlert() {
+        try {
+            Alert alert = driver.switchTo().alert();
+            String alertText = alert.getText();
+            log.info("{} - 예상치 못한 Alert 처리: {}", getSiteName(), alertText);
+            alert.accept();
+        } catch (NoAlertPresentException e) {
+            // Alert이 없으면 무시
+        } catch (Exception e) {
+            log.warn("{} - Alert 처리 중 오류: {}", getSiteName(), e.getMessage());
         }
     }
 

--- a/src/main/java/com/helpcentercrawl/crawler/impl/BusanAdminCrawler.java
+++ b/src/main/java/com/helpcentercrawl/crawler/impl/BusanAdminCrawler.java
@@ -5,6 +5,8 @@ import com.helpcentercrawl.crawler.core.AbstractCrawler;
 import com.helpcentercrawl.crawler.model.LoginModel;
 import com.helpcentercrawl.crawler.service.CrawlResultService;
 import lombok.extern.slf4j.Slf4j;
+import org.openqa.selenium.Alert;
+import org.openqa.selenium.NoAlertPresentException;
 import org.openqa.selenium.TimeoutException;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
@@ -48,7 +50,10 @@ public class BusanAdminCrawler extends AbstractCrawler {
                 .username(valueSettings.getBusanAdminUsername())
                 .password(valueSettings.getBusanAdminPassword())
                 .jsLogin(false)
-                .successCondition(ExpectedConditions.urlContains("main"))
+                .successCondition(ExpectedConditions.or(
+                        ExpectedConditions.urlContains("main"),
+                        ExpectedConditions.urlContains("changePassword")
+                ))
                 .build();
     }
 
@@ -58,14 +63,37 @@ public class BusanAdminCrawler extends AbstractCrawler {
     }
 
     @Override
-    protected void handlePopup() {
+    protected void handlePopup() throws InterruptedException {
         try {
+            // 첫 번째 Alert 처리
             WebDriverWait alertWait = new WebDriverWait(driver, Duration.ofSeconds(3));
             alertWait.until(ExpectedConditions.alertIsPresent());
 
-            driver.switchTo().alert().accept();
-        } catch (TimeoutException ignored) {
-            log.debug("Alert 창이 나타나지 않았습니다.");
+            Alert firstAlert = driver.switchTo().alert();
+            String firstAlertText = firstAlert.getText();
+            log.info("부산행정 - 첫 번째 Alert: {}", firstAlertText);
+            firstAlert.accept();
+
+            Thread.sleep(500); // 잠시 대기
+
+            // 두 번째 Alert 처리
+            try {
+                alertWait.until(ExpectedConditions.alertIsPresent());
+                Alert secondAlert = driver.switchTo().alert();
+                String secondAlertText = secondAlert.getText();
+                log.info("부산행정 - 두 번째 Alert: {}", secondAlertText);
+                secondAlert.accept();
+
+                log.info("부산행정 - 두 개의 Alert 처리 완료");
+
+            } catch (TimeoutException e) {
+                log.debug("부산행정 - 두 번째 Alert이 나타나지 않았습니다.");
+            }
+
+        } catch (TimeoutException e) {
+            log.debug("부산행정 - Alert 창이 나타나지 않았습니다.");
+        } catch (NoAlertPresentException e) {
+            log.debug("부산행정 - Alert이 존재하지 않습니다.");
         }
     }
 

--- a/src/main/java/com/helpcentercrawl/crawler/impl/BusanAdminCrawler.java
+++ b/src/main/java/com/helpcentercrawl/crawler/impl/BusanAdminCrawler.java
@@ -65,20 +65,18 @@ public class BusanAdminCrawler extends AbstractCrawler {
     @Override
     protected void handlePopup() throws InterruptedException {
         try {
-            // 첫 번째 Alert 처리
-            WebDriverWait alertWait = new WebDriverWait(driver, Duration.ofSeconds(3));
-            alertWait.until(ExpectedConditions.alertIsPresent());
+            WebDriverWait firstAlertWait = new WebDriverWait(driver, Duration.ofSeconds(5));
+            firstAlertWait.until(ExpectedConditions.alertIsPresent());
 
             Alert firstAlert = driver.switchTo().alert();
             String firstAlertText = firstAlert.getText();
             log.info("부산행정 - 첫 번째 Alert: {}", firstAlertText);
             firstAlert.accept();
 
-            Thread.sleep(500); // 잠시 대기
-
-            // 두 번째 Alert 처리
             try {
-                alertWait.until(ExpectedConditions.alertIsPresent());
+                WebDriverWait secondAlertWait = new WebDriverWait(driver, Duration.ofSeconds(2));
+                secondAlertWait.until(ExpectedConditions.alertIsPresent());
+
                 Alert secondAlert = driver.switchTo().alert();
                 String secondAlertText = secondAlert.getText();
                 log.info("부산행정 - 두 번째 Alert: {}", secondAlertText);
@@ -91,7 +89,7 @@ public class BusanAdminCrawler extends AbstractCrawler {
             }
 
         } catch (TimeoutException e) {
-            log.debug("부산행정 - Alert 창이 나타나지 않았습니다.");
+            log.debug("부산행정 - 첫 번째 Alert 창이 나타나지 않았습니다.");
         } catch (NoAlertPresentException e) {
             log.debug("부산행정 - Alert이 존재하지 않습니다.");
         }

